---
title: "Case Study: Single cell transcriptomics of human epidermis identifies basal stem cell transition states"
author:
  - name: Kayla Interdonanto
    affiliation:
    - Roswell Park Comprehensive Cancer Center
    email: Kayla.Morrell@RoswellPark.org
output:
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
date: "2 November 2022"
package: "hcaCaseStudies 0.0.0.9005"
vignette: >
  %\VignetteIndexEntry{Case Study: Single cell transcriptomics of human epidermis identifies basal stem cell transition states }
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r internal-setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE
)
```

# Setup

Load useful packages

```{r setup, message = FALSE}
library(hcaCaseStudies)
library(hca)
library(dplyr)
library(tidyr)
```

## Project information

Set the project id

```{r project_id}
project_id <- "66d7d92a-d6c5-492c-815b-f81c7c93c984"
```

Display information about the project

```{r project-information}
project_information(project_id)
```

## Conventions

Statistics calculated locally use `local.*` for column names, e.g.,
`local.cellCount`.

# Data processing

## Project & samples

Create a filter to identify the project

```{r filter}
project_filter <- filters(projectId = list(is = project_id))
```

Find the project

```{r find_project}
p <- projects(project_filter, as = "tibble_expanded")
p |> glimpse()
```

Samples associated with the project

```{r samples}
s <- samples(project_filter, as = "tibble_expanded")
```

Columns are either variable and therefore describe sample-level
information...

```{r samples-exploration}
## informative 'specimen' or 'donorOrganisms' columns
s |>
    select(starts_with("specimens") | starts_with("donorOrganisms")) |>
    select(where(column_is_informative)) |>
    glimpse()
```

...or invariant and reflect experiment-wide annotation

```{r experiment-metadata}
s |>
    select(where(column_is_experimentwide)) |>
    distinct() |>
    glimpse()
```

## Data files

What types of files does the project have?

```{r files-available}
files1 <- files(project_filter)

files1 |>
    glimpse()

files1 |>
    count(fileFormat)

## create `files2` with just the files needed for count-matrix
## analysis

files2 <- files1 |> filter(fileFormat != "fastq.gz") 
```

Retrieve and cache locally all files needed to construct a gene x cell
count matrix.

```{r retrieve}
tar_files <-
    files2 |>
    files_download() |>
    setNames(files2 |> pull(name))
```

Parse the file names to create a tibble with local file path, file type, and
sample id.

```{r local-files}
tar_files_names <- files2 |> pull(name)

local_files <- 
    tibble(
        local.filePath = file.path(tar_files),
        local.fileType = sub(".*_([[:alpha:]]*).*", "\\1Path", tar_files_names),
        local.sampleId = sub("([^_]*).*", "\\1", tar_files_names)
    ) |>
    pivot_wider(
        names_from = local.fileType,
        names_prefix = "local.",
        values_from = "local.filePath"
    )

local_files |>
    mutate(across(ends_with("Path"), basename))
```

Extract some data from the files, e.g., cell and gene counts.

```{r file-data}
gene_and_cell_counts <-
    local_files |>
    csv_tsv_gz_count_lines("local.sampleId", "local.barcodesPath") |>
    rename(local.cellCount = local.lines) |>
    csv_tsv_gz_count_lines("local.sampleId", "local.genesPath") |>
    rename(local.geneCount = local.lines)
gene_and_cell_counts |>
    select(local.sampleId, local.geneCount, local.cellCount)
```

```{r count-matrix}
object_name <- "local.countMatrix"
if (!cache_exists(project_id, object_name)) {
    local_files |>
        mtx_count_matrix(
            "local.matrixPath", "local.genesPath", "local.barcodesPath"
        ) |>
        cache_add(project_id, object_name)
}

count_matrix <- cache_read(project_id, object_name)

class(count_matrix)[1]
dim(count_matrix)
format(object.size(count_matrix), units = "auto")
```

## Row, sample, and cell annotations

Row (gene) data

```{r row-data}
row_data <-
    local_files |>
    slice(1) |>
    pull("local.genesPath") |>
    readr::read_tsv(
        col_names = c("ensembl_id", "hugo_symbol"),
        show_col_types = FALSE
    )

row_data
```

Column (sample & cell) data

```{r sample-data}
sample_data <-
    ## from the hca sample annotation...
    s |>
    select(
        where(column_is_informative) & where(is.atomic)
    ) |>
    left_join(
        gene_and_cell_counts |>
        select(local.sampleId, local.cellCount, local.geneCount),
        by = c(samples.id = "local.sampleId")
    )

sample_data
```

```{r cell-data}
cell_barcodes <- 
    local_files |>
    pull("local.barcodesPath") |>
    lapply(readLines)

sampleId <- pull(local_files, "local.sampleId")
cell_data <- tibble(
    local.sampleId = rep(sampleId, lengths(cell_barcodes)),
    local.barcodes = unlist(cell_barcodes, use.names = FALSE)
)

cell_data

## N.B. some barcodes are duplicated 2 times
cell_data |>
    count(local.barcodes, sort = TRUE, name = "occurrences") |>
    count(occurrences, name = "n_barcodes")
```

## Experiment-wide metadata

```{r metadata}
sample_metadata <-
    s |>
    select(where(column_is_experimentwide) & !any_of(names(p))) |>
    distinct()

metadata <- bind_cols(p, sample_metadata)

metadata |>
    glimpse()
```

## Assemble into SingleCellExperiment (to come)

# Anomalies

When running the count matrix, each file either had 1 or 0 cell counts.

# sessionInfo() {.unnumbered}

```{r}
sessionInfo()
```
