<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="geo_gsm_cell_count() parses a GEO GSM file for the
number of cells present in the file.
geo_gsm_gene_count() parses a GEO GSM file for the
number of genes present in the file.
geo_gsm_count_matrix() returns a sparse matrix of
gene x cell counts."><title>Summarize GEO Files — geo_gsm_cell_count • hcaCaseStudies</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Summarize GEO Files — geo_gsm_cell_count"><meta property="og:description" content="geo_gsm_cell_count() parses a GEO GSM file for the
number of cells present in the file.
geo_gsm_gene_count() parses a GEO GSM file for the
number of genes present in the file.
geo_gsm_count_matrix() returns a sparse matrix of
gene x cell counts."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hcaCaseStudies</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9003</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hca-3c9d586e-bd26-4b46-8690-3faaa18ccf38.html">Case Study: Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</a>
    <a class="dropdown-item" href="../articles/hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.html">Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</a>
    <a class="dropdown-item" href="../articles/hca-8f630e0f-6bf9-4a04-9754-02533152a954.html">Case Study: The immune cell atlas of human neuroblastoma</a>
    <a class="dropdown-item" href="../articles/hca-9fc0064b-84ce-40a5-a768-e6eb3d364ee0.html">Case Study: Single-cell RNA-seq reveals cell type-specific molecular and genetic associations to lupus</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summarize GEO Files</h1>
      
      <div class="d-none name"><code>geo.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>geo_gsm_cell_count()</code> parses a GEO GSM file for the
number of cells present in the file.</p>
<p><code>geo_gsm_gene_count()</code> parses a GEO GSM file for the
number of genes present in the file.</p>
<p><code>geo_gsm_count_matrix()</code> returns a sparse matrix of
gene x cell counts.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geo_gsm_cell_count</span><span class="op">(</span><span class="va">.data</span>, sep <span class="op">=</span> <span class="st">"\\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geo_gsm_gene_count</span><span class="op">(</span><span class="va">.data</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">geo_gsm_count_matrix</span><span class="op">(</span><span class="va">.data</span>, reader_function <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="va"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>.data</dt>
<dd><p>a <code>files_tbl_hca</code> object (e.g., created from
<code><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">hca::files()</a></code>) containing <code>fileId</code> and <code>local.filePath</code>
columns. All rows in <code>.data</code> must refer to GSM files.</p></dd>


<dt>sep</dt>
<dd><p>character(1) single character used to separate fields in
the GSM file. Typically tab (<code>'\\t"</code>, default) or comma <code>","</code>.</p></dd>


<dt>reader_function</dt>
<dd><p>a function to read each GSM file. The
function should come from the readr package. The default
<code>read_tsv</code> is appropriate for tab-delimited CSV files; a common
alternative is <code>read_csv</code> for comma-separated GSM files.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>geo_gsm_cell_count() returna </code>.data<code>augmented by a column</code>local.cellCount` reporting the number of cells in the file.</p>


<p><code>geo_gsm_gene_count()</code> returns <code>.data</code> augmented by a
column of gene counts, <code>"local.geneCount"</code>.</p>


<p><code>geo_gsm_count_matrix()</code> returns a sparse matrix with rows
(genes) equal to the unique genes in all input files, and
columns (cells) equal to the total cell count in the GSM
files. Samples are in the order present in <code>.data</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>geo_gsm_cell_count()</code> parses the first line of the GSM
file, assuming that the line is a 'header' with gene as first
column and cell labels as subsequent columns. The number of
cells is the number of columns in the file minus 1.</p>
<p><code>geo_gsm_gene_count()</code> counts the number of lines in the
file, and reports the number of genes as this number minus one,
assuming the first line is a 'header' line.</p>
<p><code>geo_gsm_count_matrix()</code> parses each GEO GSM file in turn,
assuming that the file contains a dense matrix. The dense
matrix is converted to a sparse representation. As each file is
processed, new genes are appended to the original <code>i</code> index,
and cells are appended to the <code>j</code> index. The final result is
the concatenation of all sparse matrix
representations. Progress reports the amount of memory
currently consumed by the cummulating object.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

