<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="mtx_count_matrix() parses one or several MatrixMart
files with identical dimensions into a single dgCMatrix
object."><title>Summarize 'MatrixMart' files — mtx_count_matrix • hcaCaseStudies</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Summarize 'MatrixMart' files — mtx_count_matrix"><meta property="og:description" content="mtx_count_matrix() parses one or several MatrixMart
files with identical dimensions into a single dgCMatrix
object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hcaCaseStudies</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9004</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hca-3c9d586e-bd26-4b46-8690-3faaa18ccf38.html">Case Study: Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</a>
    <a class="dropdown-item" href="../articles/hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.html">Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</a>
    <a class="dropdown-item" href="../articles/hca-8f630e0f-6bf9-4a04-9754-02533152a954.html">Case Study: The immune cell atlas of human neuroblastoma</a>
    <a class="dropdown-item" href="../articles/hca-923d3231-7295-4184-b3f6-c3082766a8c7.html">Case Study: A Single-Cell Transcriptomic Atlas of Human Skin Aging</a>
    <a class="dropdown-item" href="../articles/hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d.html">Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains</a>
    <a class="dropdown-item" href="../articles/hca-9fc0064b-84ce-40a5-a768-e6eb3d364ee0.html">Case Study: Single-cell RNA-seq reveals cell type-specific molecular and genetic associations to lupus</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summarize 'MatrixMart' files</h1>
      
      <div class="d-none name"><code>mtx.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>mtx_count_matrix()</code> parses one or several MatrixMart
files with identical dimensions into a single <code>dgCMatrix</code>
object.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mtx_count_matrix</span><span class="op">(</span></span>
<span>  <span class="va">.data</span>,</span>
<span>  <span class="va">mtx_path</span>,</span>
<span>  genes_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  barcodes_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>.data</dt>
<dd><p><code>tibble</code> containing a column with the value of the
<code>mtx_path</code> parameter and, if non-NULL, columns with the value
of parameters <code>genes_path</code> and <code>barcodes_path</code>.</p></dd>


<dt>mtx_path</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> paths to MatrixMart files on disk.</p></dd>


<dt>genes_path</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> (optional) paths to files
containing gene (row) labels (e.g., ENSEMBL gene
identifiers). The row labels must be in the first column of the
file; fields are separated by the parameter <code>sep</code>; there can be
no column headers in the file.</p></dd>


<dt>barcodes_path</dt>
<dd><p><code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code> (optional) paths to files
containing cell (column) labels (e.g., barcodes). The column
labels must be in the first column of the file; fields are
separated by the parameter <code>sep</code>; there can be no column
headers in the file.</p></dd>


<dt>sep</dt>
<dd><p><code>character(1)</code>, default <code>"\t"</code> (tab). Field separator in
<code>genes_path</code> and <code>barcodes_path</code> files.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p><code>mtx_count_matrix()</code> returns a <code>dgCMatrix</code> sparse matrix
with rows (genes) equal to the unique genes in all input files,
and columns (cells) equal to the total cell count in the
MatrixMart files. Samples (columns) are in the order present in
<code>.data</code>.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

