<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hcaCaseStudies">
<title>Case Study: A human breast atlas integrating single-cell proteomics and transcriptomics • hcaCaseStudies</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: A human breast atlas integrating single-cell proteomics and transcriptomics">
<meta property="og:description" content="hcaCaseStudies">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hcaCaseStudies</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/case_studies_summary.html">Case Studies Summary</a>
    <a class="dropdown-item" href="../articles/hca-3c9d586e-bd26-4b46-8690-3faaa18ccf38.html">Case Study: Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</a>
    <a class="dropdown-item" href="../articles/hca-4c73d1e4-bad2-4a22-a0ba-55abbdbdcc3d.html">Case Study: Single-Cell RNA Sequencing Maps Endothelial Metabolic Plasticity in Pathological Angiogenesis</a>
    <a class="dropdown-item" href="../articles/hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.html">Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</a>
    <a class="dropdown-item" href="../articles/hca-66d7d92a-d6c5-492c-815b-f81c7c93c984.html">Case Study: Single cell transcriptomics of human epidermis identifies basal stem cell transition states</a>
    <a class="dropdown-item" href="../articles/hca-74e2ef9d-7c9f-418c-b281-7fb38f3b1571.html">Case Study: The myogenesis program drives clonal selection and drug resistance in rhabdomyosarcoma</a>
    <a class="dropdown-item" href="../articles/hca-7f351a4c-d24c-4fcd-9040-f79071b097d0.html">Case Study: Expansion of Fcγ Receptor IIIa-Positive Macrophages, Ficolin 1-Positive Monocyte-Derived Dendritic Cells, and Plasmacytoid Dendritic Cells Associated With Severe Skin Disease in Systemic Sclerosis</a>
    <a class="dropdown-item" href="../articles/hca-8b9cb6ae-6a43-4e47-b9fb-3df7aeec941f.html">Case Study: Defining Transcriptional Signatures of Human Hair Follicle Cell States</a>
    <a class="dropdown-item" href="../articles/hca-8f630e0f-6bf9-4a04-9754-02533152a954.html">Case Study: The immune cell atlas of human neuroblastoma</a>
    <a class="dropdown-item" href="../articles/hca-923d3231-7295-4184-b3f6-c3082766a8c7.html">Case Study: A Single-Cell Transcriptomic Atlas of Human Skin Aging</a>
    <a class="dropdown-item" href="../articles/hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d.html">Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains</a>
    <a class="dropdown-item" href="../articles/hca-9ac53858-606a-4b89-af49-804ccedaa660.html">Case Study: A Single-Cell Characterization of Human Post-implantation Embryos Cultured In Vitro Delineates Morphogenesis in Primary Syncytialization</a>
    <a class="dropdown-item" href="../articles/hca-9b876d31-0739-4e96-9846-f76e6a427279.html">Case Study: A human breast atlas integrating single-cell proteomics and transcriptomics</a>
    <a class="dropdown-item" href="../articles/hca-9fc0064b-84ce-40a5-a768-e6eb3d364ee0.html">Case Study: Single-cell RNA-seq reveals cell type-specific molecular and genetic associations to lupus</a>
    <a class="dropdown-item" href="../articles/hca-a7c66eb1-4a4e-4f6c-9e30-ad2a485f8301.html">Case Study: Single nucleus RNA-sequencing defines unexpected diversity of cholinergic neuron types in the adult mouse spinal cord</a>
    <a class="dropdown-item" href="../articles/hca-ac289b77-fb12-4a6b-ad43-c0721c698e70.html">Case Study: Robust temporal map of human in vitro myelopoiesis using single-cell genomics</a>
    <a class="dropdown-item" href="../articles/hca-fcaa53cd-ba57-4bfe-af9c-eaa958f95c1a.html">Case Study: Mapping the developing human immune system across organs</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mtmorgan/hcaCaseStudies/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Case Study: A human breast atlas integrating single-cell proteomics and transcriptomics</h1>
                        <h4 data-toc-skip class="author">Kayla
Interdonanto</h4>
            <address class="author_afil">
      Roswell Park Comprehensive Cancer
Center<br><a class="author_email" href="mailto:#"></a><a href="mailto:Kayla.Morrell@RoswellPark.org" class="email">Kayla.Morrell@RoswellPark.org</a>
      </address>
                  
            <h4 data-toc-skip class="date">31 October 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/hcaCaseStudies/blob/HEAD/vignettes/hca-9b876d31-0739-4e96-9846-f76e6a427279.Rmd" class="external-link"><code>vignettes/hca-9b876d31-0739-4e96-9846-f76e6a427279.Rmd</code></a></small>
      <div class="d-none name"><code>hca-9b876d31-0739-4e96-9846-f76e6a427279.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Load useful packages</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/hcaCaseStudies/">hcaCaseStudies</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">hca</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="project-information">Project information<a class="anchor" aria-label="anchor" href="#project-information"></a>
</h3>
<p>Set the project id</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_id</span> <span class="op">&lt;-</span> <span class="st">"9b876d31-0739-4e96-9846-f76e6a427279"</span></span></code></pre></div>
<p>Display information about the project</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/hca/man/project_information.html" class="external-link">project_information</a></span><span class="op">(</span><span class="va">project_id</span><span class="op">)</span></span>
<span><span class="co">## Title</span></span>
<span><span class="co">##   A human breast atlas integrating single-cell proteomics and</span></span>
<span><span class="co">##   transcriptomics</span></span>
<span><span class="co">## Contributors (unknown order; any role)</span></span>
<span><span class="co">##   Jia-Ren Lin, Peter K Sorger, Aviv Regev, Rachel Schwartz, Laura M</span></span>
<span><span class="co">##   Selfors, Kurt D'Andrea, Sandro Santagata, Walter L Goh, Deborah A</span></span>
<span><span class="co">##   Dillon, Jennifer M Rosenbluth, Ron C J Schackmann, Carman Man-Chung</span></span>
<span><span class="co">##   Li, Shaolin Mei, Katherine L Nathanson, Kaitlin Moore, G Kenneth</span></span>
<span><span class="co">##   Gray, Nomeda Girnius, Judy E Garber, Joan S Brugge, Hana K Shapiro</span></span>
<span><span class="co">## Description</span></span>
<span><span class="co">##   The breast is a dynamic organ whose response to physiological and</span></span>
<span><span class="co">##   pathophysiological conditions alters its disease susceptibility, yet</span></span>
<span><span class="co">##   the specific effects of these clinical variables on cell state remain</span></span>
<span><span class="co">##   poorly annotated. We present a unified, high-resolution breast atlas</span></span>
<span><span class="co">##   by integrating single-cell RNA-seq, mass cytometry, and cyclic</span></span>
<span><span class="co">##   immunofluorescence, encompassing a myriad of states. We define cell</span></span>
<span><span class="co">##   subtypes within the alveolar, hormone-sensing, and basal epithelial</span></span>
<span><span class="co">##   lineages, delineating associations of several subtypes with cancer</span></span>
<span><span class="co">##   risk factors, including age, parity, and BRCA2 germline mutation. Of</span></span>
<span><span class="co">##   particular interest is a subset of alveolar cells termed</span></span>
<span><span class="co">##   basal-luminal (BL) cells, which exhibit poor transcriptional lineage</span></span>
<span><span class="co">##   fidelity, accumulate with age, and carry a gene signature associated</span></span>
<span><span class="co">##   with basal-like breast cancer. We further utilize a medium-depletion</span></span>
<span><span class="co">##   approach to identify molecular factors regulating cell-subtype</span></span>
<span><span class="co">##   proportion in organoids. Together, these data are a rich resource to</span></span>
<span><span class="co">##   elucidate diverse mammary cell states. Overall design: A total of 16</span></span>
<span><span class="co">##   breast samples were assayed (4 samples from reductive mammoplasties</span></span>
<span><span class="co">##   and 12 from prophylactic mastectomies).</span></span>
<span><span class="co">## DOI</span></span>
<span><span class="co">##   10.1016/j.devcel.2022.05.003</span></span>
<span><span class="co">## URL</span></span>
<span><span class="co">##   https://pubmed.ncbi.nlm.nih.gov/35617956/</span></span>
<span><span class="co">## Project</span></span>
<span><span class="co">##   https://data.humancellatlas.org/explore/projects/9b876d31-0739-4e96-9846-f76e6a427279</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="conventions">Conventions<a class="anchor" aria-label="anchor" href="#conventions"></a>
</h3>
<p>Statistics calculated locally use <code>local.*</code> for column
names, e.g., <code>local.cellCount</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-processing">Data processing<a class="anchor" aria-label="anchor" href="#data-processing"></a>
</h2>
<div class="section level3">
<h3 id="project-samples">Project &amp; samples<a class="anchor" aria-label="anchor" href="#project-samples"></a>
</h3>
<p>Create a filter to identify the project</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span>projectId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="va">project_id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Find the project</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span><span class="va">project_filter</span>, as <span class="op">=</span> <span class="st">"tibble_expanded"</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 69</span></span>
<span><span class="co">## $ projectId                                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9b876d31-0739-4e96-984…</span></span>
<span><span class="co">## $ cellSuspensions.organ                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ cellSuspensions.organPart                      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ cellSuspensions.selectedCellType               <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ cellSuspensions.totalCells                     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 120000</span></span>
<span><span class="co">## $ dates.aggregateLastModifiedDate                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-23T17:25:02.58…</span></span>
<span><span class="co">## $ dates.aggregateSubmissionDate                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-05-31T17:34:57.20…</span></span>
<span><span class="co">## $ dates.aggregateUpdateDate                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-23T17:25:02.58…</span></span>
<span><span class="co">## $ dates.lastModifiedDate                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-23T12:48:58.23…</span></span>
<span><span class="co">## $ dates.submissionDate                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-05-31T17:34:57.20…</span></span>
<span><span class="co">## $ dates.updateDate                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-23T12:48:58.23…</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "female"</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.disease                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ donorOrganisms.donorCount                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 16</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ donorOrganisms.id                              <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Individual_01", "Indi…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> ["25.0", "year", "31.0"…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.unit                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"year", "year", "year"…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.value               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"25.0", "31.0", "33.0"…</span></span>
<span><span class="co">## $ donorOrganisms.organismAgeRange                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;788400000, 788400000, …</span></span>
<span><span class="co">## $ fileTypeSummaries.contentDescription           <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"DNA sequence", "Cell …</span></span>
<span><span class="co">## $ fileTypeSummaries.count                        <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;57, 1, 1&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ fileTypeSummaries.format                       <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"fastq.gz", "csv.gz", …</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate               <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ fileTypeSummaries.matrixCellCount              <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ fileTypeSummaries.totalSize                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;493925635516, 319972, …</span></span>
<span><span class="co">## $ projects.accessible                            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ projects.accessions.accession                  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"GSE180878", "SRP32997…</span></span>
<span><span class="co">## $ projects.accessions.namespace                  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"geo_series", "insdc_…</span></span>
<span><span class="co">## $ projects.contributors.contactName              <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Jia-Ren,,Lin", "Pete…</span></span>
<span><span class="co">## $ projects.contributors.correspondingContributor <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ projects.contributors.email                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"raschwar@ucsc.edu", …</span></span>
<span><span class="co">## $ projects.contributors.institution              <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"The Laboratory of Sy…</span></span>
<span><span class="co">## $ projects.contributors.laboratory               <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ projects.contributors.projectRole              <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"data curator", "expe…</span></span>
<span><span class="co">## $ projects.estimatedCellCount                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 52682</span></span>
<span><span class="co">## $ projects.laboratory                            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ projects.projectDescription                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "The breast is a dynam…</span></span>
<span><span class="co">## $ projects.projectId                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9b876d31-0739-4e96-98…</span></span>
<span><span class="co">## $ projects.projectShortname                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breastTranscriptomeAtl…</span></span>
<span><span class="co">## $ projects.projectTitle                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A human breast atlas …</span></span>
<span><span class="co">## $ projects.publications.doi                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10.1016/j.devcel.2022.…</span></span>
<span><span class="co">## $ projects.publications.officialHcaPublication   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ projects.publications.publicationTitle         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A human breast atlas i…</span></span>
<span><span class="co">## $ projects.publications.publicationUrl           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://pubmed.ncbi.n…</span></span>
<span><span class="co">## $ projects.supplementaryLinks                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina HiSeq X"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v2"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single cell"</span></span>
<span><span class="co">## $ protocols.pairedEnd                            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"analysis_protocol_1",…</span></span>
<span><span class="co">## $ samples.disease                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ samples.effectiveOrgan                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ samples.id                                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"SAMN20422990", "SAMN…</span></span>
<span><span class="co">## $ samples.organ                                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ samples.organPart                              <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ samples.preservationMethod                     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ samples.sampleEntityType                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span>
<span><span class="co">## $ sources.sourceId                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "de02b927-8c8a-45fd-ad…</span></span>
<span><span class="co">## $ sources.sourceSpec                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-6d6c8d6e…</span></span>
<span><span class="co">## $ specimens.disease                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ specimens.id                                   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"SAMN20422990", "SAMN2…</span></span>
<span><span class="co">## $ specimens.organ                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ specimens.organPart                            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ specimens.preservationMethod                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ specimens.source                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span></code></pre></div>
<p>Samples associated with the project</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/samples.html" class="external-link">samples</a></span><span class="op">(</span><span class="va">project_filter</span>, as <span class="op">=</span> <span class="st">"tibble_expanded"</span><span class="op">)</span></span></code></pre></div>
<p>Columns are either variable and therefore describe sample-level
information…</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## informative 'specimen' or 'donorOrganisms' columns</span></span>
<span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"specimens"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"donorOrganisms"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_informative</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 16</span></span>
<span><span class="co">## Columns: 5</span></span>
<span><span class="co">## $ specimens.id                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "SAMN20422998", "SAMN20423002", "SAMN…</span></span>
<span><span class="co">## $ donorOrganisms.id                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Individual_05", "Individual_01", "In…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.unit  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "year", "year", "year", "year", "year…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.value <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "25.0", "36.0", "52.0", "53.0", "39.0…</span></span>
<span><span class="co">## $ donorOrganisms.organismAgeRange  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;788400000, 788400000&gt;, &lt;1135296000,…</span></span></code></pre></div>
<p>…or invariant and reflect experiment-wide annotation</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_experimentwide</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 28</span></span>
<span><span class="co">## $ cellSuspensions.organ                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ cellSuspensions.totalCells            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 7500</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "female"</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.disease                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ donorOrganisms.donorCount             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.estimatedCellCount           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 52682</span></span>
<span><span class="co">## $ projects.projectId                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9b876d31-0739-4e96-9846-f76e6a4…</span></span>
<span><span class="co">## $ projects.projectShortname             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breastTranscriptomeAtlas"</span></span>
<span><span class="co">## $ projects.projectTitle                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A human breast atlas integratin…</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina HiSeq X"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v2"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single cell"</span></span>
<span><span class="co">## $ protocols.pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"analysis_protocol_1", "analysi…</span></span>
<span><span class="co">## $ samples.accessible                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ samples.disease                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ samples.effectiveOrgan                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ samples.organ                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ samples.sampleEntityType              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span>
<span><span class="co">## $ sources.sourceId                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "de02b927-8c8a-45fd-adef-c801e32…</span></span>
<span><span class="co">## $ sources.sourceSpec                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-6d6c8d6e:snapshot/…</span></span>
<span><span class="co">## $ specimens.disease                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal"</span></span>
<span><span class="co">## $ specimens.organ                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "breast"</span></span>
<span><span class="co">## $ specimens.source                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-files">Data files<a class="anchor" aria-label="anchor" href="#data-files"></a>
</h3>
<p>What types of files does the project have?</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">files1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files</a></span><span class="op">(</span><span class="va">project_filter</span><span class="op">)</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 59</span></span>
<span><span class="co">## Columns: 8</span></span>
<span><span class="co">## $ fileId       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "d26e7a79-babe-4e53-8529-4e8024306c3a", "958de4b8-0da1-46…</span></span>
<span><span class="co">## $ name         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "SRR15254362_I1.fastq.gz", "SRR15254369_R1.fastq.gz", "SR…</span></span>
<span><span class="co">## $ fileFormat   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "fastq.gz", "fastq.gz", "fastq.gz", "fastq.gz", "fastq.gz…</span></span>
<span><span class="co">## $ size         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 4473288809, 3589069190, 3970703806, 9335218336, 435259777…</span></span>
<span><span class="co">## $ version      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-07-26T00:16:46.719000Z", "2022-07-26T00:16:47.15200…</span></span>
<span><span class="co">## $ projectTitle <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "A human breast atlas integrating single-cell proteomics …</span></span>
<span><span class="co">## $ projectId    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9b876d31-0739-4e96-9846-f76e6a427279", "9b876d31-0739-4e…</span></span>
<span><span class="co">## $ url          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://service.azul.data.humancellatlas.org/repository/…</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">fileFormat</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">##   fileFormat     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> csv.gz         2</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> fastq.gz      57</span></span>
<span></span>
<span><span class="co">## create `files2` with just the files needed for count-matrix</span></span>
<span><span class="co">## analysis</span></span>
<span></span>
<span><span class="va">files2</span> <span class="op">&lt;-</span> <span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"csv.gz"</span><span class="op">)</span></span></code></pre></div>
<p>Retrieve and cache locally all files needed to construct a gene x
cell count matrix.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file_path</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files2</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files_download</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">files2</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## join file path to appropriate components of `f`</span></span>
<span><span class="va">count_matrix_files1</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files2</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>        fileId <span class="op">=</span> <span class="va">files2</span><span class="op">$</span><span class="va">fileId</span>,</span>
<span>        local.filePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">file_path</span><span class="op">)</span></span>
<span>    <span class="op">)</span>, by <span class="op">=</span> <span class="st">"fileId"</span><span class="op">)</span></span></code></pre></div>
<p>We can extract some data from the files, e.g., cell and gene
counts</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 'files' is the final, locally modified, version of file data</span></span>
<span><span class="va">count_matrix_files</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">count_matrix_files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">name</span> <span class="op">==</span> <span class="st">"GSE180878_Li_Brugge_10XscRNAseq_GeneCellMatrix_RNAcounts_human.csv.gz"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/geo.html">cell_count_csv</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/geo.html">gene_count</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">count_matrix_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">local.geneCount</span>, <span class="va">local.cellCount</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">##   name                                                           local…¹ local…²</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> GSE180878_Li_Brugge_10XscRNAseq_GeneCellMatrix_RNAcounts_huma…   <span style="text-decoration: underline;">20</span>437   <span style="text-decoration: underline;">52</span>681</span></span>
<span><span class="co">## <span style="color: #949494;"># … with abbreviated variable names ¹​local.geneCount, ²​local.cellCount</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="count-matrix">Count matrix<a class="anchor" aria-label="anchor" href="#count-matrix"></a>
</h3>
<p>Create a <code>dgCMatrix</code> sparse matrix (defined in the
<code>Matrix</code> package). Save it in the package cache for re-use
across <em>R</em> sessions.</p>
<!--
add `eval = !nzchar(Sys.getenv("GITHUB_ACTIONS"))` if requires
more than 7 GB
-->
</div>
</div>
<div class="section level2">
<h2 id="anomalies">Anomalies<a class="anchor" aria-label="anchor" href="#anomalies"></a>
</h2>
<p>No contributor generated matrices and analysis files for this
project.</p>
<p>Ran into a ‘VROOM_CONNECTION_SIZE’ error. I increaded the size and
reran code only to have R crash. I believe the file is too large to
process on my computer.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="sessioninfo">sessionInfo()<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R Under development (unstable) (2022-12-10 r83428)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin21.6.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Users/ma38727/bin/R-devel/lib/libRblas.dylib </span></span>
<span><span class="co">## LAPACK: /Users/ma38727/bin/R-devel/lib/libRlapack.dylib;  LAPACK version 3.10.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] dplyr_1.0.10              hca_1.7.1                </span></span>
<span><span class="co">## [3] hcaCaseStudies_0.0.0.9006 BiocStyle_2.27.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] xfun_0.35           bslib_0.4.1         htmlwidgets_1.5.4  </span></span>
<span><span class="co">##  [4] lattice_0.20-45     tzdb_0.3.0          vctrs_0.5.1        </span></span>
<span><span class="co">##  [7] tools_4.3.0         generics_0.1.3      curl_4.3.3         </span></span>
<span><span class="co">## [10] tibble_3.1.8        fansi_1.0.3         RSQLite_2.2.19     </span></span>
<span><span class="co">## [13] blob_1.2.3          pkgconfig_2.0.3     Matrix_1.5-3       </span></span>
<span><span class="co">## [16] dbplyr_2.2.1        desc_1.4.2          assertthat_0.2.1   </span></span>
<span><span class="co">## [19] lifecycle_1.0.3     compiler_4.3.0      stringr_1.5.0      </span></span>
<span><span class="co">## [22] textshaping_0.3.6   codetools_0.2-18    httpuv_1.6.6       </span></span>
<span><span class="co">## [25] htmltools_0.5.4     sass_0.4.4          yaml_2.3.6         </span></span>
<span><span class="co">## [28] whisker_0.4.1       tidyr_1.2.1         pillar_1.8.1       </span></span>
<span><span class="co">## [31] pkgdown_2.0.6.9000  later_1.3.0         jquerylib_0.1.4    </span></span>
<span><span class="co">## [34] ellipsis_0.3.2      DT_0.26             cachem_1.0.6       </span></span>
<span><span class="co">## [37] mime_0.12           tidyselect_1.2.0    digest_0.6.30      </span></span>
<span><span class="co">## [40] stringi_1.7.8       purrr_0.3.5         bookdown_0.30      </span></span>
<span><span class="co">## [43] rprojroot_2.0.3     fastmap_1.1.0       grid_4.3.0         </span></span>
<span><span class="co">## [46] cli_3.4.1           magrittr_2.0.3      utf8_1.2.2         </span></span>
<span><span class="co">## [49] withr_2.5.0         readr_2.1.3         filelock_1.0.2     </span></span>
<span><span class="co">## [52] promises_1.2.0.1    bit64_4.0.5         rmarkdown_2.18     </span></span>
<span><span class="co">## [55] httr_1.4.4          bit_4.0.5           hms_1.1.2          </span></span>
<span><span class="co">## [58] ragg_1.2.4          memoise_2.0.1       shiny_1.7.3        </span></span>
<span><span class="co">## [61] evaluate_0.18       knitr_1.41          miniUI_0.1.1.1     </span></span>
<span><span class="co">## [64] BiocFileCache_2.7.1 rlang_1.0.6         Rcpp_1.0.9         </span></span>
<span><span class="co">## [67] xtable_1.8-4        glue_1.6.2          DBI_1.1.3          </span></span>
<span><span class="co">## [70] BiocManager_1.30.19 jsonlite_1.8.4      R6_2.5.1           </span></span>
<span><span class="co">## [73] systemfonts_1.0.4   fs_1.5.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Kayla Interdonanto.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
