<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hcaCaseStudies">
<title>Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains • hcaCaseStudies</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains">
<meta property="og:description" content="hcaCaseStudies">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hcaCaseStudies</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9005</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hca-3c9d586e-bd26-4b46-8690-3faaa18ccf38.html">Case Study: Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</a>
    <a class="dropdown-item" href="../articles/hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.html">Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</a>
    <a class="dropdown-item" href="../articles/hca-8f630e0f-6bf9-4a04-9754-02533152a954.html">Case Study: The immune cell atlas of human neuroblastoma</a>
    <a class="dropdown-item" href="../articles/hca-923d3231-7295-4184-b3f6-c3082766a8c7.html">Case Study: A Single-Cell Transcriptomic Atlas of Human Skin Aging</a>
    <a class="dropdown-item" href="../articles/hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d.html">Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains</a>
    <a class="dropdown-item" href="../articles/hca-9fc0064b-84ce-40a5-a768-e6eb3d364ee0.html">Case Study: Single-cell RNA-seq reveals cell type-specific molecular and genetic associations to lupus</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/datatables-binding-0.25/datatables.js"></script><link href="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains</h1>
                        <h4 data-toc-skip class="author">Martin
Morgan</h4>
            <address class="author_afil">
      Roswell Park Comprehensive Cancer
Center<br><a class="author_email" href="mailto:#"></a><a href="mailto:Martin.Morgan@RoswellPark.org" class="email">Martin.Morgan@RoswellPark.org</a>
      </address>
                  
            <h4 data-toc-skip class="date">4 October 2022</h4>
      
      
      <div class="d-none name"><code>hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Load useful packages</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/hcaCaseStudies/">hcaCaseStudies</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">hca</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="va">pivot_wider</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="va"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span></span></code></pre></div>
<div class="section level3">
<h3 id="project-information">Project information<a class="anchor" aria-label="anchor" href="#project-information"></a>
</h3>
<p>Set the project id</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_id</span> <span class="op">&lt;-</span> <span class="st">"9a23ac2d-93dd-4bac-9bb8-040e6426db9d"</span></span></code></pre></div>
<p>Display information about the project</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/project.html">project_information</a></span><span class="op">(</span><span class="va">project_id</span><span class="op">)</span></span>
<span><span class="co">## Title</span></span>
<span><span class="co">##   Single-cell transcriptomic and proteomic analysis of Parkinson’s</span></span>
<span><span class="co">##   disease brains</span></span>
<span><span class="co">## Contributors (unknown order; any role)</span></span>
<span><span class="co">##   Le Zhang, Rachel Schwartz</span></span>
<span><span class="co">## Description</span></span>
<span><span class="co">##   Parkinson’s disease (PD) is a prevalent neurodegenerative disorder</span></span>
<span><span class="co">##   where recent evidence suggests pathogenesis may be mediated by</span></span>
<span><span class="co">##   inflammatory processes. The molecular architecture of the disease</span></span>
<span><span class="co">##   remains to be fully elucidated. We performed single-nucleus</span></span>
<span><span class="co">##   transcriptomics and unbiased proteomics using postmortem tissue</span></span>
<span><span class="co">##   obtained from the prefrontal cortex of 12 individuals with late-stage</span></span>
<span><span class="co">##   PD and age-matched controls. We analyzed ~80,000 nuclei and</span></span>
<span><span class="co">##   identified eight major cell types, including brain-resident T cells,</span></span>
<span><span class="co">##   each with distinct transcriptional changes in line with the known</span></span>
<span><span class="co">##   genetics of PD. By analyzing Lewy body pathology in the same</span></span>
<span><span class="co">##   postmortem tissue, we found that α-synuclein pathology is inversely</span></span>
<span><span class="co">##   correlated with chaperone expression in excitatory neurons. Examining</span></span>
<span><span class="co">##   cell-cell interactions, we found a selective abatement of</span></span>
<span><span class="co">##   neuron-astrocyte interactions and enhanced neuroinflammation.</span></span>
<span><span class="co">##   Proteomic analyses of the same brains identified synaptic proteins in</span></span>
<span><span class="co">##   prefrontal cortex that were preferentially downregulated in PD.</span></span>
<span><span class="co">##   Strikingly, comparing this dataset to a regionally similar published</span></span>
<span><span class="co">##   analysis for Alzheimer’s disease (AD), we found no common</span></span>
<span><span class="co">##   differentially expressed genes in neurons, but identified many shared</span></span>
<span><span class="co">##   differentially expressed genes in glial cells, suggesting that</span></span>
<span><span class="co">##   disease etiology in PD and AD are likely distinct. These data are</span></span>
<span><span class="co">##   presented as a resource for interrogating the molecular and cellular</span></span>
<span><span class="co">##   basis of PD and other neurodegenerative diseases.</span></span>
<span><span class="co">## doi: NA</span></span>
<span><span class="co">## url: NA</span></span>
<span><span class="co">## project: https://data.humancellatlas.org/explore/projects/9a23ac2d-93dd-4bac-9bb8-040e6426db9d</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="conventions">Conventions<a class="anchor" aria-label="anchor" href="#conventions"></a>
</h3>
<p>Statistics calculated locally use <code>local.*</code> for column
names, e.g., <code>local.cellCount</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-processing">Data processing<a class="anchor" aria-label="anchor" href="#data-processing"></a>
</h2>
<div class="section level3">
<h3 id="project-samples">Project &amp; samples<a class="anchor" aria-label="anchor" href="#project-samples"></a>
</h3>
<p>Create a filter to identify the project</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span>projectId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="va">project_id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Find the project</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span><span class="va">project_filter</span>, as <span class="op">=</span> <span class="st">"tibble_expanded"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co">## some column names are very long...</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename_with</a></span><span class="op">(</span></span>
<span>        <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".*\\.(.*)"</span>, <span class="st">"projects.contributedAnalyses.\\1"</span>, <span class="va">.x</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"projects.contributedAnalyses"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 76</span></span>
<span><span class="co">## $ projectId                                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9b…</span></span>
<span><span class="co">## $ cellSuspensions.organ                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ cellSuspensions.organPart                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ cellSuspensions.selectedCellType                <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ cellSuspensions.totalCells                      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ dates.aggregateLastModifiedDate                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T23:02:34.0…</span></span>
<span><span class="co">## $ dates.aggregateSubmissionDate                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T19:50:13.3…</span></span>
<span><span class="co">## $ dates.aggregateUpdateDate                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T23:02:34.0…</span></span>
<span><span class="co">## $ dates.lastModifiedDate                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:12.5…</span></span>
<span><span class="co">## $ dates.submissionDate                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T19:50:13.3…</span></span>
<span><span class="co">## $ dates.updateDate                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:12.5…</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"female", "male"&gt;</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.disease                          <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease", …</span></span>
<span><span class="co">## $ donorOrganisms.donorCount                       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 12</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ donorOrganisms.id                               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"HSDG07HC_HHT", "HSDG…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.unit                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"year", "year", "year…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.value                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"63.0", "65.0", "69.…</span></span>
<span><span class="co">## $ donorOrganisms.organismAgeRange                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;1986768000, 198676800…</span></span>
<span><span class="co">## $ fileTypeSummaries.contentDescription            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Feature table", "Ge…</span></span>
<span><span class="co">## $ fileTypeSummaries.count                         <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;1, 36&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ fileTypeSummaries.format                        <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"tar", "fastq.gz"&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate                <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ fileTypeSummaries.matrixCellCount               <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ fileTypeSummaries.totalSize                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;719482880, 5061288652…</span></span>
<span><span class="co">## $ projects.accessible                             <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ projects.accessions.accession                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GSE202210"</span></span>
<span><span class="co">## $ projects.accessions.namespace                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "geo_series"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.contentDescription <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Feature table", "Gen…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.drs_uri            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "drs://data.terra.bio…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.fileSource         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.format             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tar"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.isIntermediate     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.contributedAnalyses.matrixCellCount    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ projects.contributedAnalyses.name               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GSE202210_RAW.tar"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.sha256             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "6d65eb60c292d5657e1b…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.size               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 719482880</span></span>
<span><span class="co">## $ projects.contributedAnalyses.url                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://service.azul…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.uuid               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "a56df86f-a78e-4270-9…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.version            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:59.…</span></span>
<span><span class="co">## $ projects.contributors.contactName               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Le,,Zhang", "Rachel…</span></span>
<span><span class="co">## $ projects.contributors.correspondingContributor  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;TRUE, FALSE&gt;</span></span>
<span><span class="co">## $ projects.contributors.email                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"le.zhang@yale.edu", …</span></span>
<span><span class="co">## $ projects.contributors.institution               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Yale University", "…</span></span>
<span><span class="co">## $ projects.contributors.laboratory                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Neurology", "Human C…</span></span>
<span><span class="co">## $ projects.contributors.projectRole               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"principal investigat…</span></span>
<span><span class="co">## $ projects.estimatedCellCount                     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 80000</span></span>
<span><span class="co">## $ projects.laboratory                             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Human Cell Atlas Da…</span></span>
<span><span class="co">## $ projects.projectDescription                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Parkinson’s disease (…</span></span>
<span><span class="co">## $ projects.projectId                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9b…</span></span>
<span><span class="co">## $ projects.projectShortname                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "ZhangLabPdBrainNuclei"</span></span>
<span><span class="co">## $ projects.projectTitle                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell transcrip…</span></span>
<span><span class="co">## $ projects.supplementaryLinks                     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina NovaSeq 6000"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single nucleus"</span></span>
<span><span class="co">## $ protocols.pairedEnd                             <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ samples.disease                                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease",…</span></span>
<span><span class="co">## $ samples.effectiveOrgan                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.id                                      <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"GSM6106340", "GSM610…</span></span>
<span><span class="co">## $ samples.organ                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.organPart                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ samples.preservationMethod                      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ samples.sampleEntityType                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organis…</span></span>
<span><span class="co">## $ sources.sourceId                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "aa623d49-c868-4a3a-83…</span></span>
<span><span class="co">## $ sources.sourceSpec                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-c341e9ae…</span></span>
<span><span class="co">## $ specimens.disease                               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease", …</span></span>
<span><span class="co">## $ specimens.id                                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"GSM6106340", "GSM610…</span></span>
<span><span class="co">## $ specimens.organ                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ specimens.organPart                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ specimens.preservationMethod                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ specimens.source                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organis…</span></span></code></pre></div>
<p>Samples associated with the project</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/samples.html" class="external-link">samples</a></span><span class="op">(</span><span class="va">project_filter</span>, as <span class="op">=</span> <span class="st">"tibble_expanded"</span><span class="op">)</span></span></code></pre></div>
<p>Columns are either variable and therefore describe sample-level
information…</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## informative 'specimen' or 'donorOrganisms' columns</span></span>
<span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"specimens"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"donorOrganisms"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_informative</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 12 × 7</span></span></span>
<span><span class="co">##    specimens.disease specimens.id donorOrganis…¹ donor…² donor…³ donor…⁴ donor…⁵</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> normal            GSM6106343   female         normal  hsDG13… 65.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> Parkinson disease GSM6106348   male           Parkin… HSDG19… 86.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> normal            GSM6106340   female         normal  HSDG07… 81.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> Parkinson disease GSM6106347   female         Parkin… hsDG19… 72.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> normal            GSM6106350   male           normal  hsDG30… 63.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> Parkinson disease GSM6106346   female         Parkin… hsDG15… 81.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> normal            GSM6106351   male           normal  hsDG99… 69.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> Parkinson disease GSM6106349   female         Parkin… hsDG20… 77.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> normal            GSM6106342   male           normal  HSDG10… 70.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> Parkinson disease GSM6106345   male           Parkin… HSDG14… 96.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> normal            GSM6106341   female         normal  hsDG10… 72.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> Parkinson disease GSM6106344   male           Parkin… hsDG14… 79.0    &lt;dbl&gt;  </span></span>
<span><span class="co">## <span style="color: #949494;"># … with abbreviated variable names ¹​donorOrganisms.biologicalSex,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​donorOrganisms.disease, ³​donorOrganisms.id,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ⁴​donorOrganisms.organismAge.value, ⁵​donorOrganisms.organismAgeRange</span></span></span>
<span></span>
<span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">specimens.disease</span>, <span class="va">donorOrganisms.biologicalSex</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">pivot_wider</span><span class="op">(</span></span>
<span>        names_from <span class="op">=</span> <span class="st">"donorOrganisms.biologicalSex"</span>,</span>
<span>        values_from <span class="op">=</span> <span class="st">"n"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2 × 3</span></span></span>
<span><span class="co">##   specimens.disease female  male</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> normal                 3     3</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> Parkinson disease      3     3</span></span></code></pre></div>
<p>…or invariant and reflect experiment-wide annotation (also found in
the project data)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_experimentwide</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 32</span></span>
<span><span class="co">## $ cellSuspensions.organ                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ cellSuspensions.organPart             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefrontal cortex"</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.donorCount             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.unit       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "year"</span></span>
<span><span class="co">## $ fileTypeSummaries.contentDescription  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Feature table", "Gene expressi…</span></span>
<span><span class="co">## $ fileTypeSummaries.count               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;1, 3&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ fileTypeSummaries.format              <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"tar", "fastq.gz"&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.estimatedCellCount           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 80000</span></span>
<span><span class="co">## $ projects.laboratory                   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Human Cell Atlas Data Coordina…</span></span>
<span><span class="co">## $ projects.projectId                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9bb8-040e642…</span></span>
<span><span class="co">## $ projects.projectShortname             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "ZhangLabPdBrainNuclei"</span></span>
<span><span class="co">## $ projects.projectTitle                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell transcriptomic and …</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina NovaSeq 6000"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single nucleus"</span></span>
<span><span class="co">## $ protocols.pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ samples.accessible                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ samples.effectiveOrgan                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.organ                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.organPart                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefrontal cortex"</span></span>
<span><span class="co">## $ samples.sampleEntityType              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span>
<span><span class="co">## $ sources.sourceId                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "aa623d49-c868-4a3a-8365-bf895f0…</span></span>
<span><span class="co">## $ sources.sourceSpec                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-c341e9ae:snapshot/…</span></span>
<span><span class="co">## $ specimens.organ                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ specimens.organPart                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefrontal cortex"</span></span>
<span><span class="co">## $ specimens.source                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-files">Data files<a class="anchor" aria-label="anchor" href="#data-files"></a>
</h3>
<p>What types of files does the project have?</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">files1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files</a></span><span class="op">(</span><span class="va">project_filter</span><span class="op">)</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 37</span></span>
<span><span class="co">## Columns: 8</span></span>
<span><span class="co">## $ fileId       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "47c37cc9-72ea-473e-a2fc-2e92c3263069", "eaf79c9e-1d18-47…</span></span>
<span><span class="co">## $ name         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "SRR19085688_I1.fastq.gz", "SRR19085694_R1.fastq.gz", "SR…</span></span>
<span><span class="co">## $ fileFormat   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "fastq.gz", "fastq.gz", "fastq.gz", "fastq.gz", "fastq.gz…</span></span>
<span><span class="co">## $ size         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> 5545686966, 12335195016, 5789934692, 10301311199, 2458048…</span></span>
<span><span class="co">## $ version      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:23.041000Z", "2022-08-26T21:58:23.40100…</span></span>
<span><span class="co">## $ projectTitle <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell transcriptomic and proteomic analysis of Par…</span></span>
<span><span class="co">## $ projectId    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9bb8-040e6426db9d", "9a23ac2d-93dd-4b…</span></span>
<span><span class="co">## $ url          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://service.azul.data.humancellatlas.org/repository/…</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">fileFormat</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">##   fileFormat     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> fastq.gz      36</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> tar            1</span></span></code></pre></div>
<p>Looks like a ‘tar’ file contains useful data</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">files2</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"tar"</span><span class="op">)</span></span></code></pre></div>
<p>Retrieve and cache locally all files needed to construct a gene x
cell count matrix.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tar_file</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files2</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files_download</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="va">files2</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">name</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>This is a ‘tar’ archive of GSM files; untar these to a temporary
location</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## what files are in the tar?</span></span>
<span><span class="va">tar_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">tar_file</span>, list <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">tar_files</span><span class="op">)</span></span>
<span><span class="co">## [1] 36</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">tar_files</span><span class="op">)</span></span>
<span><span class="co">## [1] "GSM6106340_HSDG07HC_barcodes.tsv.gz" </span></span>
<span><span class="co">## [2] "GSM6106340_HSDG07HC_genes.tsv.gz"    </span></span>
<span><span class="co">## [3] "GSM6106340_HSDG07HC_matrix.mtx.gz"   </span></span>
<span><span class="co">## [4] "GSM6106341_hsDG101HC_barcodes.tsv.gz"</span></span>
<span><span class="co">## [5] "GSM6106341_hsDG101HC_genes.tsv.gz"   </span></span>
<span><span class="co">## [6] "GSM6106341_hsDG101HC_matrix.mtx.gz"</span></span>
<span></span>
<span><span class="co">## untar to a temporary location</span></span>
<span><span class="va">exdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">exdir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">tar_file</span>, exdir <span class="op">=</span> <span class="va">exdir</span><span class="op">)</span></span></code></pre></div>
<p>Parse the file names to create a tibble with local file path, file
type, and sample id</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">local_files</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>        local.filePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">exdir</span>, <span class="va">tar_files</span><span class="op">)</span>,</span>
<span>        <span class="co">## local.fileType becomes one of barcodesPath, genesPath, matrixPath</span></span>
<span>        local.fileType <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">".*_([[:alpha:]]*).*"</span>, <span class="st">"\\1Path"</span>, <span class="va">tar_files</span><span class="op">)</span>,</span>
<span>        local.sampleId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"([^_]*).*"</span>, <span class="st">"\\1"</span>, <span class="va">tar_files</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co">## each 'sampleId' has a barcodesPath, genesPath, and matrixPath</span></span>
<span>    <span class="fu">pivot_wider</span><span class="op">(</span></span>
<span>        names_from <span class="op">=</span> <span class="va">local.fileType</span>,</span>
<span>        names_prefix <span class="op">=</span> <span class="st">"local."</span>,</span>
<span>        values_from <span class="op">=</span> <span class="st">"local.filePath"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">## 12 samples</span></span>
<span><span class="va">local_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">ends_with</a></span><span class="op">(</span><span class="st">"Path"</span><span class="op">)</span>, <span class="va">basename</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 12 × 4</span></span></span>
<span><span class="co">##    local.sampleId local.barcodesPath                   local.genesPath   local…¹</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> GSM6106340     GSM6106340_HSDG07HC_barcodes.tsv.gz  GSM6106340_HSDG0… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> GSM6106341     GSM6106341_hsDG101HC_barcodes.tsv.gz GSM6106341_hsDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> GSM6106342     GSM6106342_HSDG10HC_barcodes.tsv.gz  GSM6106342_HSDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> GSM6106343     GSM6106343_hsDG13HC_barcodes.tsv.gz  GSM6106343_hsDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> GSM6106344     GSM6106344_hsDG142PD_barcodes.tsv.gz GSM6106344_hsDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> GSM6106345     GSM6106345_HSDG148PD_barcodes.tsv.gz GSM6106345_HSDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> GSM6106346     GSM6106346_hsDG151PD_barcodes.tsv.gz GSM6106346_hsDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> GSM6106347     GSM6106347_hsDG197PD_barcodes.tsv.gz GSM6106347_hsDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> GSM6106348     GSM6106348_HSDG199PD_barcodes.tsv.gz GSM6106348_HSDG1… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> GSM6106349     GSM6106349_hsDG208PD_barcodes.tsv.gz GSM6106349_hsDG2… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> GSM6106350     GSM6106350_hsDG30HC_barcodes.tsv.gz  GSM6106350_hsDG3… GSM610…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> GSM6106351     GSM6106351_hsDG99HC_barcodes.tsv.gz  GSM6106351_hsDG9… GSM610…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with abbreviated variable name ¹​local.matrixPath</span></span></span></code></pre></div>
<p>Extract some data from the files, e.g., cell and gene counts.</p>
<!--
add `eval = !nzchar(Sys.getenv("GITHUB_ACTIONS"))` if requires
more than 7 GB
-->
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_and_cell_counts</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">local_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/csv_tsv.html">csv_tsv_gz_count_lines</a></span><span class="op">(</span><span class="st">"local.sampleId"</span>, <span class="st">"local.barcodesPath"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>local.cellCount <span class="op">=</span> <span class="va">local.lines</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/csv_tsv.html">csv_tsv_gz_count_lines</a></span><span class="op">(</span><span class="st">"local.sampleId"</span>, <span class="st">"local.genesPath"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>local.geneCount <span class="op">=</span> <span class="va">local.lines</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gene_and_cell_counts</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">local.sampleId</span>, <span class="va">local.geneCount</span>, <span class="va">local.cellCount</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 12 × 3</span></span></span>
<span><span class="co">##    local.sampleId local.geneCount local.cellCount</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> GSM6106340               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">4</span>813</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> GSM6106341               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">5</span>863</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> GSM6106342               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">7</span>455</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> GSM6106343               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">5</span>092</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> GSM6106344               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">8</span>316</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> GSM6106345               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">9</span>071</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> GSM6106346               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">1</span>891</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> GSM6106347               <span style="text-decoration: underline;">33</span>538           <span style="text-decoration: underline;">10</span>440</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> GSM6106348               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">7</span>365</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> GSM6106349               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">5</span>633</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span> GSM6106350               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">6</span>246</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">12</span> GSM6106351               <span style="text-decoration: underline;">33</span>538            <span style="text-decoration: underline;">5</span>798</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="count-matrix">Count matrix<a class="anchor" aria-label="anchor" href="#count-matrix"></a>
</h3>
<p>Each sample has a ‘matrix mart’ text file representing counts in all
cells of the sample. Use <code><a href="../reference/mtx.html">mtx_count_matrix()</a></code> to read these
into a <code>dgCMatrix</code> sparse matrix (defined in the
<code>Matrix</code> package). Save it in the package cache for re-use
across <em>R</em> sessions.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object_name</span> <span class="op">&lt;-</span> <span class="st">"local.countMatrix"</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="../reference/cache.html">cache_exists</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">local_files</span> <span class="op">|&gt;</span></span>
<span>        <span class="fu"><a href="../reference/mtx.html">mtx_count_matrix</a></span><span class="op">(</span></span>
<span>            <span class="st">"local.matrixPath"</span>, <span class="st">"local.genesPath"</span>, <span class="st">"local.barcodesPath"</span></span>
<span>        <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>        <span class="fu"><a href="../reference/cache.html">cache_add</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cache.html">cache_read</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="co"># sparse matrix</span></span>
<span><span class="co">## [1] "dgCMatrix"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>, units <span class="op">=</span> <span class="st">"auto"</span><span class="op">)</span> <span class="co"># how much memory?</span></span>
<span><span class="co">## [1] "2.6 Gb"</span></span>
<span></span>
<span><span class="co">## dimensions of genes x cells count matrix should match annotation summaries</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>      <span class="co"># genes x cells</span></span>
<span><span class="co">## [1] 33538 77983</span></span>
<span><span class="va">gene_and_cell_counts</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">local.geneCount</span><span class="op">)</span>, cells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">local.cellCount</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 2</span></span></span>
<span><span class="co">##   genes cells</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> <span style="text-decoration: underline;">33</span>538 <span style="text-decoration: underline;">77</span>983</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="row-sample-and-cell-annotations">Row, sample, and cell annotations<a class="anchor" aria-label="anchor" href="#row-sample-and-cell-annotations"></a>
</h3>
<p>Row (gene) data</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## all 'genes' files are identical?</span></span>
<span><span class="va">row_data</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">local_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="st">"local.genesPath"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span></span>
<span>        col_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ensembl_id"</span>, <span class="st">"hugo_symbol"</span>, <span class="st">"annotation"</span><span class="op">)</span>,</span>
<span>        show_col_types <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>        <span class="op">)</span></span>
<span></span>
<span><span class="va">row_data</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 33,538 × 3</span></span></span>
<span><span class="co">##    ensembl_id      hugo_symbol annotation     </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> ENSG00000243485 MIR1302-2HG Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> ENSG00000237613 FAM138A     Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> ENSG00000186092 OR4F5       Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> ENSG00000238009 AL627309.1  Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> ENSG00000239945 AL627309.3  Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> ENSG00000239906 AL627309.2  Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> ENSG00000241599 AL627309.4  Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> ENSG00000236601 AL732372.1  Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> ENSG00000284733 OR4F29      Gene Expression</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> ENSG00000235146 AC114498.1  Gene Expression</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 33,528 more rows</span></span></span></code></pre></div>
<p>Column (sample &amp; cell) data</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_data</span> <span class="op">&lt;-</span></span>
<span>    <span class="co">## from the hca sample annotation...</span></span>
<span>    <span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>        <span class="fu">where</span><span class="op">(</span><span class="va">column_is_informative</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu">where</span><span class="op">(</span><span class="va">is.atomic</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co">## ...and local_files (gene and cell counts)</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="va">gene_and_cell_counts</span> <span class="op">|&gt;</span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">local.sampleId</span>, <span class="va">local.cellCount</span>, <span class="va">local.geneCount</span><span class="op">)</span>,</span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>samples.id <span class="op">=</span> <span class="st">"local.sampleId"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">sample_data</span> <span class="op">|&gt;</span> <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-6f8820d8034b173d54f6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-6f8820d8034b173d54f6">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["01287b66-cb26-4543-87b7-3773b161203d","08071351-dc2a-46c1-8ad1-9c8711e4133b","0bc73d79-bfcd-44e3-b2f1-304989cc9997","0ef3b775-f29f-4c96-8a5c-d75d5394e032","20a56799-0790-4824-a737-64a23e1dd4b8","47e1e803-d60e-4299-b3e3-3eb2409f6958","83a6ffc8-5e71-4bab-bedf-0716be9e75e9","83d2996a-a5f2-409a-9788-3f43cc5efc25","aab34bad-2ce3-4545-a95b-2494e70bb40c","ad64ebbb-c133-4cfb-b315-6dd1fb4def7d","dce22b60-669b-4d98-b4d6-094de3e228c6","dd14ce46-a73c-468d-9297-9806e027f19b"],["2022-08-26T22:41:05.457000Z","2022-08-26T22:41:05.487000Z","2022-08-26T22:41:05.437000Z","2022-08-26T22:41:05.482000Z","2022-08-26T22:41:05.501000Z","2022-08-26T22:41:05.476000Z","2022-08-26T22:41:05.508000Z","2022-08-26T22:41:05.494000Z","2022-08-26T22:41:05.450000Z","2022-08-26T22:41:05.470000Z","2022-08-26T22:41:05.444000Z","2022-08-26T22:41:05.463000Z"],["2022-08-26T21:58:22.549000Z","2022-08-26T21:58:22.629000Z","2022-08-26T21:58:22.503000Z","2022-08-26T21:58:22.614000Z","2022-08-26T21:58:22.662000Z","2022-08-26T21:58:22.599000Z","2022-08-26T21:58:22.676000Z","2022-08-26T21:58:22.648000Z","2022-08-26T21:58:22.535000Z","2022-08-26T21:58:22.582000Z","2022-08-26T21:58:22.521000Z","2022-08-26T21:58:22.566000Z"],["2022-08-26T22:41:05.457000Z","2022-08-26T22:41:05.487000Z","2022-08-26T22:41:05.437000Z","2022-08-26T22:41:05.482000Z","2022-08-26T22:41:05.501000Z","2022-08-26T22:41:05.476000Z","2022-08-26T22:41:05.508000Z","2022-08-26T22:41:05.494000Z","2022-08-26T22:41:05.450000Z","2022-08-26T22:41:05.470000Z","2022-08-26T22:41:05.444000Z","2022-08-26T22:41:05.463000Z"],["female","male","female","female","male","female","male","female","male","male","female","male"],["normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease"],["hsDG13HC_HHT","HSDG199PD_HHT","HSDG07HC_HHT","hsDG197PD_HHT","hsDG30HC_HHT","hsDG151PD_HHT","hsDG99HC_HHT","hsDG208PD_HHT","HSDG10HC_HHT","HSDG148PD_HHT","hsDG101HC_HHT","hsDG142PD_HHT"],["65.0","86.0","81.0","72.0","63.0","81.0","69.0","77.0","70.0","96.0","72.0","79.0"],["normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease"],["GSM6106343","GSM6106348","GSM6106340","GSM6106347","GSM6106350","GSM6106346","GSM6106351","GSM6106349","GSM6106342","GSM6106345","GSM6106341","GSM6106344"],["normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease","normal","Parkinson disease"],["GSM6106343","GSM6106348","GSM6106340","GSM6106347","GSM6106350","GSM6106346","GSM6106351","GSM6106349","GSM6106342","GSM6106345","GSM6106341","GSM6106344"],[5092,7365,4813,10440,6246,1891,5798,5633,7455,9071,5863,8316],[33538,33538,33538,33538,33538,33538,33538,33538,33538,33538,33538,33538]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>entryId<\/th>\n      <th>dates.lastModifiedDate<\/th>\n      <th>dates.submissionDate<\/th>\n      <th>dates.updateDate<\/th>\n      <th>donorOrganisms.biologicalSex<\/th>\n      <th>donorOrganisms.disease<\/th>\n      <th>donorOrganisms.id<\/th>\n      <th>donorOrganisms.organismAge.value<\/th>\n      <th>samples.disease<\/th>\n      <th>samples.id<\/th>\n      <th>specimens.disease<\/th>\n      <th>specimens.id<\/th>\n      <th>local.cellCount<\/th>\n      <th>local.geneCount<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[13,14]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cell_barcodes</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">local_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="st">"local.barcodesPath"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">readLines</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sampleId</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">local_files</span>, <span class="st">"local.sampleId"</span><span class="op">)</span></span>
<span><span class="va">cell_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>    local.sampleId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">sampleId</span>, <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">cell_barcodes</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    local.barcodes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">cell_barcodes</span>, use.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">cell_data</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 77,983 × 2</span></span></span>
<span><span class="co">##    local.sampleId local.barcodes    </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> GSM6106340     AAACCCAAGAATAACC-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> GSM6106340     AAACCCAAGTCATCCA-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> GSM6106340     AAACCCAGTAATCAGA-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> GSM6106340     AAACCCATCCATTCGC-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> GSM6106340     AAACGAAAGGAGTACC-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> GSM6106340     AAACGAACATGAAGCG-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> GSM6106340     AAACGAAGTGCCTACG-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> GSM6106340     AAACGAAGTGTTCATG-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> GSM6106340     AAACGAAGTTGCCAAT-1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> GSM6106340     AAACGAAGTTGCTGAT-1</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 77,973 more rows</span></span></span>
<span></span>
<span><span class="co">## N.B. some barcodes are duplicated 2 or 3 times</span></span>
<span><span class="va">cell_data</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">local.barcodes</span>, sort <span class="op">=</span> <span class="cn">TRUE</span>, name <span class="op">=</span> <span class="st">"occurrences"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">occurrences</span>, name <span class="op">=</span> <span class="st">"n_barcodes"</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">##   occurrences n_barcodes</span></span>
<span><span class="co">##         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span>           1      <span style="text-decoration: underline;">76</span>272</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span>           2        830</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span>           3         17</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="experiment-wide-metadata">Experiment-wide metadata<a class="anchor" aria-label="anchor" href="#experiment-wide-metadata"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample_metadata</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_experimentwide</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">any_of</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_cols</a></span><span class="op">(</span><span class="va">p</span>, <span class="va">sample_metadata</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metadata</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 77</span></span>
<span><span class="co">## $ projectId                                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9b…</span></span>
<span><span class="co">## $ cellSuspensions.organ                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ cellSuspensions.organPart                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ cellSuspensions.selectedCellType                <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ cellSuspensions.totalCells                      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ dates.aggregateLastModifiedDate                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T23:02:34.0…</span></span>
<span><span class="co">## $ dates.aggregateSubmissionDate                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T19:50:13.3…</span></span>
<span><span class="co">## $ dates.aggregateUpdateDate                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T23:02:34.0…</span></span>
<span><span class="co">## $ dates.lastModifiedDate                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:12.5…</span></span>
<span><span class="co">## $ dates.submissionDate                            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T19:50:13.3…</span></span>
<span><span class="co">## $ dates.updateDate                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:12.5…</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"female", "male"&gt;</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.disease                          <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease", …</span></span>
<span><span class="co">## $ donorOrganisms.donorCount                       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 12</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ donorOrganisms.id                               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"HSDG07HC_HHT", "HSDG…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.unit                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"year", "year", "year…</span></span>
<span><span class="co">## $ donorOrganisms.organismAge.value                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"63.0", "65.0", "69.…</span></span>
<span><span class="co">## $ donorOrganisms.organismAgeRange                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;1986768000, 198676800…</span></span>
<span><span class="co">## $ fileTypeSummaries.contentDescription            <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Feature table", "Ge…</span></span>
<span><span class="co">## $ fileTypeSummaries.count                         <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;1, 36&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ fileTypeSummaries.format                        <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"tar", "fastq.gz"&gt;</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate                <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ fileTypeSummaries.matrixCellCount               <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ fileTypeSummaries.totalSize                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;719482880, 5061288652…</span></span>
<span><span class="co">## $ projects.accessible                             <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ projects.accessions.accession                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GSE202210"</span></span>
<span><span class="co">## $ projects.accessions.namespace                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "geo_series"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.contentDescription <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Feature table", "Gen…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.drs_uri            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "drs://data.terra.bio…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.fileSource         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.format             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tar"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.isIntermediate     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.contributedAnalyses.matrixCellCount    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ projects.contributedAnalyses.name               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GSE202210_RAW.tar"</span></span>
<span><span class="co">## $ projects.contributedAnalyses.sha256             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "6d65eb60c292d5657e1b…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.size               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 719482880</span></span>
<span><span class="co">## $ projects.contributedAnalyses.url                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "https://service.azul…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.uuid               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "a56df86f-a78e-4270-9…</span></span>
<span><span class="co">## $ projects.contributedAnalyses.version            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "2022-08-26T21:58:59.…</span></span>
<span><span class="co">## $ projects.contributors.contactName               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Le,,Zhang", "Rachel…</span></span>
<span><span class="co">## $ projects.contributors.correspondingContributor  <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;TRUE, FALSE&gt;</span></span>
<span><span class="co">## $ projects.contributors.email                     <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"le.zhang@yale.edu", …</span></span>
<span><span class="co">## $ projects.contributors.institution               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Yale University", "…</span></span>
<span><span class="co">## $ projects.contributors.laboratory                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Neurology", "Human C…</span></span>
<span><span class="co">## $ projects.contributors.projectRole               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"principal investigat…</span></span>
<span><span class="co">## $ projects.estimatedCellCount                     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 80000</span></span>
<span><span class="co">## $ projects.laboratory                             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Human Cell Atlas Da…</span></span>
<span><span class="co">## $ projects.projectDescription                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Parkinson’s disease (…</span></span>
<span><span class="co">## $ projects.projectId                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "9a23ac2d-93dd-4bac-9b…</span></span>
<span><span class="co">## $ projects.projectShortname                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "ZhangLabPdBrainNuclei"</span></span>
<span><span class="co">## $ projects.projectTitle                           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell transcrip…</span></span>
<span><span class="co">## $ projects.supplementaryLinks                     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina NovaSeq 6000"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single nucleus"</span></span>
<span><span class="co">## $ protocols.pairedEnd                             <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ samples.disease                                 <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease",…</span></span>
<span><span class="co">## $ samples.effectiveOrgan                          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.id                                      <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"GSM6106340", "GSM610…</span></span>
<span><span class="co">## $ samples.organ                                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ samples.organPart                               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ samples.preservationMethod                      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ samples.sampleEntityType                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                                  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organis…</span></span>
<span><span class="co">## $ sources.sourceId                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "aa623d49-c868-4a3a-83…</span></span>
<span><span class="co">## $ sources.sourceSpec                              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-c341e9ae…</span></span>
<span><span class="co">## $ specimens.disease                               <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"Parkinson disease", …</span></span>
<span><span class="co">## $ specimens.id                                    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"GSM6106340", "GSM610…</span></span>
<span><span class="co">## $ specimens.organ                                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "brain"</span></span>
<span><span class="co">## $ specimens.organPart                             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "dorsolateral prefront…</span></span>
<span><span class="co">## $ specimens.preservationMethod                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ specimens.source                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organis…</span></span>
<span><span class="co">## $ samples.accessible                              <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="assemble-into-singlecellexperiment-to-come">Assemble into SingleCellExperiment (to come)<a class="anchor" aria-label="anchor" href="#assemble-into-singlecellexperiment-to-come"></a>
</h3>
</div>
</div>
<div class="section level2">
<h2 id="anomalies">Anomalies<a class="anchor" aria-label="anchor" href="#anomalies"></a>
</h2>
<p>None noted.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="sessioninfo">sessionInfo()<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.2.1 Patched (2022-10-03 r83008)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin21.6.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Users/ma38727/bin/R-4-2-branch/lib/libRblas.dylib</span></span>
<span><span class="co">## LAPACK: /Users/ma38727/bin/R-4-2-branch/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] dplyr_1.0.10              hca_1.5.7                </span></span>
<span><span class="co">## [3] hcaCaseStudies_0.0.0.9005 BiocStyle_2.25.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.9          lattice_0.20-45     tidyr_1.2.1        </span></span>
<span><span class="co">##  [4] assertthat_0.2.1    rprojroot_2.0.3     digest_0.6.29      </span></span>
<span><span class="co">##  [7] utf8_1.2.2          mime_0.12           BiocFileCache_2.5.0</span></span>
<span><span class="co">## [10] R6_2.5.1            RSQLite_2.2.18      evaluate_0.16      </span></span>
<span><span class="co">## [13] httr_1.4.4          pillar_1.8.1        rlang_1.0.6        </span></span>
<span><span class="co">## [16] curl_4.3.2          miniUI_0.1.1.1      whisker_0.4        </span></span>
<span><span class="co">## [19] jquerylib_0.1.4     blob_1.2.3          Matrix_1.5-1       </span></span>
<span><span class="co">## [22] DT_0.25             rmarkdown_2.16      pkgdown_2.0.6      </span></span>
<span><span class="co">## [25] textshaping_0.3.6   desc_1.4.2          readr_2.1.3        </span></span>
<span><span class="co">## [28] stringr_1.4.1       htmlwidgets_1.5.4   bit_4.0.4          </span></span>
<span><span class="co">## [31] shiny_1.7.2         compiler_4.2.1      httpuv_1.6.6       </span></span>
<span><span class="co">## [34] xfun_0.33           pkgconfig_2.0.3     systemfonts_1.0.4  </span></span>
<span><span class="co">## [37] htmltools_0.5.3     tidyselect_1.1.2    tibble_3.1.8       </span></span>
<span><span class="co">## [40] bookdown_0.29       fansi_1.0.3         crayon_1.5.2       </span></span>
<span><span class="co">## [43] tzdb_0.3.0          dbplyr_2.2.1        later_1.3.0        </span></span>
<span><span class="co">## [46] rappdirs_0.3.3      grid_4.2.1          jsonlite_1.8.2     </span></span>
<span><span class="co">## [49] xtable_1.8-4        lifecycle_1.0.2     DBI_1.1.3          </span></span>
<span><span class="co">## [52] magrittr_2.0.3      vroom_1.6.0         cli_3.4.1          </span></span>
<span><span class="co">## [55] stringi_1.7.8       cachem_1.0.6        fs_1.5.2           </span></span>
<span><span class="co">## [58] promises_1.2.0.1    bslib_0.4.0         ellipsis_0.3.2     </span></span>
<span><span class="co">## [61] filelock_1.0.2      ragg_1.2.3          generics_0.1.3     </span></span>
<span><span class="co">## [64] vctrs_0.4.2         tools_4.2.1         bit64_4.0.5        </span></span>
<span><span class="co">## [67] glue_1.6.2          purrr_0.3.5         crosstalk_1.2.0    </span></span>
<span><span class="co">## [70] hms_1.1.2           parallel_4.2.1      fastmap_1.1.0      </span></span>
<span><span class="co">## [73] yaml_2.3.5          BiocManager_1.30.18 memoise_2.0.1      </span></span>
<span><span class="co">## [76] knitr_1.40          sass_0.4.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
