<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="hcaCaseStudies">
<title>Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum • hcaCaseStudies</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum">
<meta property="og:description" content="hcaCaseStudies">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">hcaCaseStudies</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9005</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/hca-3c9d586e-bd26-4b46-8690-3faaa18ccf38.html">Case Study: Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets</a>
    <a class="dropdown-item" href="../articles/hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.html">Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</a>
    <a class="dropdown-item" href="../articles/hca-8f630e0f-6bf9-4a04-9754-02533152a954.html">Case Study: The immune cell atlas of human neuroblastoma</a>
    <a class="dropdown-item" href="../articles/hca-923d3231-7295-4184-b3f6-c3082766a8c7.html">Case Study: A Single-Cell Transcriptomic Atlas of Human Skin Aging</a>
    <a class="dropdown-item" href="../articles/hca-9a23ac2d-93dd-4bac-9bb8-040e6426db9d.html">Case Study: Single-cell transcriptomic and proteomic analysis of Parkinson’s disease brains</a>
    <a class="dropdown-item" href="../articles/hca-9fc0064b-84ce-40a5-a768-e6eb3d364ee0.html">Case Study: Single-cell RNA-seq reveals cell type-specific molecular and genetic associations to lupus</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/datatables-binding-0.25/datatables.js"></script><link href="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Case Study: Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</h1>
                        <h4 data-toc-skip class="author">Martin
Morgan</h4>
            <address class="author_afil">
      Roswell Park Comprehensive Cancer
Center<br><a class="author_email" href="mailto:#"></a><a href="mailto:Martin.Morgan@RoswellPark.org" class="email">Martin.Morgan@RoswellPark.org</a>
      </address>
                  
            <h4 data-toc-skip class="date">23 September 2022</h4>
      
      
      <div class="d-none name"><code>hca-5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Load useful packages</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/hcaCaseStudies/">hcaCaseStudies</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">hca</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="project-information">Project information<a class="anchor" aria-label="anchor" href="#project-information"></a>
</h3>
<p>Set the project id</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_id</span> <span class="op">&lt;-</span> <span class="st">"5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776"</span></span></code></pre></div>
<p>Display information about the project</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/project.html">project_information</a></span><span class="op">(</span><span class="va">project_id</span><span class="op">)</span></span>
<span><span class="co">## Title</span></span>
<span><span class="co">##   Single-cell Transcriptome Atlas of the Human Corpus Cavernosum</span></span>
<span><span class="co">## Contributors (unknown order; any role)</span></span>
<span><span class="co">##   Zheng Li, Brittney,D,Wick</span></span>
<span><span class="co">## Description</span></span>
<span><span class="co">##   Three normal tissue samples were obtained from the tumour margin of</span></span>
<span><span class="co">##   penile carcinoma resection, all patients reported good stimulated</span></span>
<span><span class="co">##   erections and early morning erections. The five ED CC tissues were</span></span>
<span><span class="co">##   obtained from biopsy samples in artificial cavernous body</span></span>
<span><span class="co">##   implantation, all the ED patients were diagnosed with organic ED</span></span>
<span><span class="co">##   rather than psychological ED by the nocturnal penile tumescence (NPT)</span></span>
<span><span class="co">##   and intracavernosal injection (ICI) tests. The DMED patients had a at</span></span>
<span><span class="co">##   least 10-year history of type 1 diabetes and had good blood glucose</span></span>
<span><span class="co">##   control before surgery.</span></span>
<span><span class="co">## doi: NA</span></span>
<span><span class="co">## url: NA</span></span>
<span><span class="co">## project: https://data.humancellatlas.org/explore/projects/5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="conventions">Conventions<a class="anchor" aria-label="anchor" href="#conventions"></a>
</h3>
<p>Statistics calculated locally use <code>local.*</code> for column
names, e.g., <code>local.cellCount</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="data-processing">Data processing<a class="anchor" aria-label="anchor" href="#data-processing"></a>
</h2>
<div class="section level3">
<h3 id="project-samples">Project &amp; samples<a class="anchor" aria-label="anchor" href="#project-samples"></a>
</h3>
<p>Create a filter to identify the project</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">project_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/filters.html" class="external-link">filters</a></span><span class="op">(</span>projectId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>is <span class="op">=</span> <span class="va">project_id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Find the project</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/projects.html" class="external-link">projects</a></span><span class="op">(</span><span class="va">project_filter</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 14</span></span>
<span><span class="co">## $ projectId                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "5b910a43-7fb5-4ea7-b9d6-43dbd1bf2776"</span></span>
<span><span class="co">## $ projectTitle                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell Transcriptome Atlas of the Hu…</span></span>
<span><span class="co">## $ genusSpecies                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ sampleEntityType            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ specimenOrgan               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ specimenOrganPart           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ selectedCellType            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> NA</span></span>
<span><span class="co">## $ libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single cell"</span></span>
<span><span class="co">## $ pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ workflow                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ specimenDisease             <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"erectile dysfunction", "normal"&gt;</span></span>
<span><span class="co">## $ donorDisease                <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> &lt;"diabetes", "erectile dysfunction", "norm…</span></span>
<span><span class="co">## $ developmentStage            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span></code></pre></div>
<p>Samples associated with the project</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/samples.html" class="external-link">samples</a></span><span class="op">(</span><span class="va">project_filter</span>, as <span class="op">=</span> <span class="st">"tibble_expanded"</span><span class="op">)</span></span></code></pre></div>
<p>Columns are either variable and therefore describe sample-level
information…</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## informative 'specimen' or 'donorOrganisms' columns</span></span>
<span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"specimens"</span><span class="op">)</span> <span class="op">|</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">starts_with</a></span><span class="op">(</span><span class="st">"donorOrganisms"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_informative</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 8</span></span>
<span><span class="co">## Columns: 4</span></span>
<span><span class="co">## $ specimens.disease      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "normal", "erectile dysfunction", "normal", "er…</span></span>
<span><span class="co">## $ specimens.id           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GSM6255907", "GSM6255912", "GSM6255909", "GSM6…</span></span>
<span><span class="co">## $ donorOrganisms.disease <span style="color: #949494; font-style: italic;">&lt;list&gt;</span> "normal", "erectile dysfunction", "normal", "e…</span></span>
<span><span class="co">## $ donorOrganisms.id      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "donor_1", "donor_6", "donor_3", "donor_4", "do…</span></span>
<span></span>
<span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">specimens.id</span>, <span class="va">specimens.disease</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 8 × 3</span></span></span>
<span><span class="co">##   specimens.id specimens.disease        n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> GSM6255907   normal                   1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> GSM6255908   normal                   1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> GSM6255909   normal                   1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> GSM6255910   erectile dysfunction     1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> GSM6255911   erectile dysfunction     1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> GSM6255912   erectile dysfunction     1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> GSM6255913   erectile dysfunction     1</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">8</span> GSM6255914   erectile dysfunction     1</span></span></code></pre></div>
<p>…or invariant and reflect experiment-wide annotation</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_experimentwide</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 29</span></span>
<span><span class="co">## $ cellSuspensions.organ                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ cellSuspensions.organPart             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "male"</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.donorCount             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.estimatedCellCount           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 37892</span></span>
<span><span class="co">## $ projects.laboratory                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Human Cell Atlas Data Coordinat…</span></span>
<span><span class="co">## $ projects.projectId                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "5b910a43-7fb5-4ea7-b9d6-43dbd1b…</span></span>
<span><span class="co">## $ projects.projectShortname             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "AtlasOfTheHumanCorpusCavernosum"</span></span>
<span><span class="co">## $ projects.projectTitle                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell Transcriptome Atlas…</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina NovaSeq 6000"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single cell"</span></span>
<span><span class="co">## $ protocols.pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ samples.accessible                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ samples.effectiveOrgan                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ samples.organ                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ samples.organPart                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ samples.sampleEntityType              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span>
<span><span class="co">## $ sources.sourceId                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "3ab24677-3fed-42cd-b686-b82b294…</span></span>
<span><span class="co">## $ sources.sourceSpec                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-465c9e6a:snapshot/…</span></span>
<span><span class="co">## $ specimens.organ                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ specimens.organPart                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ specimens.source                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-files">Data files<a class="anchor" aria-label="anchor" href="#data-files"></a>
</h3>
<p>What types of files does the project have?</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">files1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files</a></span><span class="op">(</span><span class="va">project_filter</span><span class="op">)</span></span>
<span></span>
<span><span class="va">files1</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 23 × 8</span></span></span>
<span><span class="co">##    fileId                    name  fileF…¹    size version proje…² proje…³ url  </span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 48879b15-921c-49c0-907e-… SRR1… fastq.… 6.65<span style="color: #949494;">e</span> 9 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> d6a8d77c-0039-41c9-b5a8-… SRR1… fastq.… 2.37<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 76506d2b-4fe2-4130-ba99-… SRR1… fastq.… 2.23<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 41ee77b8-d908-49cd-bb2f-… SRR1… fastq.… 6.09<span style="color: #949494;">e</span> 9 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> f74c196a-8f82-4312-9646-… GSE2… tar     1.77<span style="color: #949494;">e</span> 8 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 896cc476-88f0-42fe-97d9-… SRR1… fastq.… 2.33<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 26eada48-7446-4e7b-8891-… SRR1… fastq.… 2.52<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 5d4e02e5-da34-44ab-b2dd-… SRR1… fastq.… 2.40<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 34488f52-d477-429a-94be-… SRR1… fastq.… 4.79<span style="color: #949494;">e</span> 9 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> d39a1587-addf-4dd3-8ff9-… SRR1… fastq.… 2.42<span style="color: #949494;">e</span>10 2022-0… Single… 5b910a… http…</span></span>
<span><span class="co">## <span style="color: #949494;"># … with 13 more rows, and abbreviated variable names ¹​fileFormat,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   ²​projectTitle, ³​projectId</span></span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">fileFormat</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">##   fileFormat     n</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> fastq.gz      22</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> tar            1</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">fileFormat</span> <span class="op">==</span> <span class="st">"tar"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">fileId</span>, <span class="va">name</span>, <span class="va">fileFormat</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 3</span></span></span>
<span><span class="co">##   fileId                               name              fileFormat</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> f74c196a-8f82-4312-9646-65655f4c0486 GSE206528_RAW.tar tar</span></span></code></pre></div>
<p>Retrieve, list the content, and untar (to a temporary location; untar
is relatively quick) the file.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tar_file</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html" class="external-link">startsWith</a></span><span class="op">(</span><span class="va">name</span>, <span class="st">"GSE"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/hca/man/files.html" class="external-link">files_download</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## what files are in the tar?</span></span>
<span><span class="va">tar_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">tar_file</span>, list <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">tar_files</span></span>
<span><span class="co">## [1] "GSM6255907_Normal_1_gene_martix.csv.gz"  </span></span>
<span><span class="co">## [2] "GSM6255908_Normal_2_gene_matrix.csv.gz"  </span></span>
<span><span class="co">## [3] "GSM6255909_Normal_3_gene_matrix.csv.gz"  </span></span>
<span><span class="co">## [4] "GSM6255910_non-DM1_gene_matrix.csv.gz"   </span></span>
<span><span class="co">## [5] "GSM6255911_non-DM2_gene_matrix.csv.gz"   </span></span>
<span><span class="co">## [6] "GSM6255912_non-DM3_gene_matrix.csv.gz"   </span></span>
<span><span class="co">## [7] "GSM6255913_Diabetes_1_gene_matrix.csv.gz"</span></span>
<span><span class="co">## [8] "GSM6255914_Diabetes_2_gene_martix.csv.gz"</span></span>
<span></span>
<span><span class="co">## untar to a temporary location</span></span>
<span><span class="va">exdir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>; <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">exdir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="va">tar_file</span>, exdir <span class="op">=</span> <span class="va">exdir</span><span class="op">)</span></span></code></pre></div>
<p>Annotate the <code>files1</code> object with local file
information</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">files_local</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span></span>
<span>    fileId <span class="op">=</span> <span class="va">files1</span><span class="op">$</span><span class="va">fileId</span>,</span>
<span>    local.fileName <span class="op">=</span> <span class="va">tar_files</span>,</span>
<span>    local.filePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">exdir</span>, <span class="va">tar_files</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">files2</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files1</span> <span class="op">|&gt;</span></span>
<span>    <span class="co">## 1:many mapping from files1 to files_local</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">right_join</a></span><span class="op">(</span><span class="va">files_local</span>, by <span class="op">=</span> <span class="st">"fileId"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="co">## make fileId unique</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>fileId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">fileId</span>, <span class="va">local.fileName</span>, sep <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We can extract some data from the files, e.g., cell and gene
counts</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">count_matrix_files</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">files2</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/geo.html">geo_gsm_cell_count</a></span><span class="op">(</span>sep <span class="op">=</span> <span class="st">","</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/geo.html">geo_gsm_gene_count</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">count_matrix_files</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">local.fileName</span>, <span class="va">local.cellCount</span>, <span class="va">local.geneCount</span><span class="op">)</span></span>
<span><span class="co">## <span style="color: #949494;"># A tibble: 8 × 3</span></span></span>
<span><span class="co">##   local.fileName                           local.cellCount local.geneCount</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                              <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> GSM6255907_Normal_1_gene_martix.csv.gz              <span style="text-decoration: underline;">8</span>357           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> GSM6255908_Normal_2_gene_matrix.csv.gz              <span style="text-decoration: underline;">5</span>571           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> GSM6255909_Normal_3_gene_matrix.csv.gz             <span style="text-decoration: underline;">11</span>644           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> GSM6255910_non-DM1_gene_matrix.csv.gz               <span style="text-decoration: underline;">5</span>906           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> GSM6255911_non-DM2_gene_matrix.csv.gz               <span style="text-decoration: underline;">2</span>853           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> GSM6255912_non-DM3_gene_matrix.csv.gz              <span style="text-decoration: underline;">11</span>179           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">7</span> GSM6255913_Diabetes_1_gene_matrix.csv.gz            <span style="text-decoration: underline;">9</span>058           <span style="text-decoration: underline;">33</span>694</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">8</span> GSM6255914_Diabetes_2_gene_martix.csv.gz           <span style="text-decoration: underline;">10</span>425           <span style="text-decoration: underline;">33</span>694</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="count-matrix">Count matrix<a class="anchor" aria-label="anchor" href="#count-matrix"></a>
</h3>
<p>Compute the count matrix by concatenating counts from the GSM
files.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object_name</span> <span class="op">&lt;-</span> <span class="st">"local.countMatrix"</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="../reference/cache.html">cache_exists</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">count_matrix_files</span> <span class="op">|&gt;</span></span>
<span>        <span class="fu"><a href="../reference/geo.html">geo_gsm_count_matrix</a></span><span class="op">(</span>reader_function <span class="op">=</span> <span class="fu">readr</span><span class="fu">::</span><span class="va"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_csv</a></span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>        <span class="fu"><a href="../reference/cache.html">cache_add</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">count_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cache.html">cache_read</a></span><span class="op">(</span><span class="va">project_id</span>, <span class="va">object_name</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="co"># sparse matrix</span></span>
<span><span class="co">## [1] "dgCMatrix"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>      <span class="co"># genes x cells</span></span>
<span><span class="co">## [1] 33694 64993</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/format.html" class="external-link">format</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/object.size.html" class="external-link">object.size</a></span><span class="op">(</span><span class="va">count_matrix</span><span class="op">)</span>, units <span class="op">=</span> <span class="st">"auto"</span><span class="op">)</span></span>
<span><span class="co">## [1] "1.8 Gb"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="singlecellexperiment">SingleCellExperiment<a class="anchor" aria-label="anchor" href="#singlecellexperiment"></a>
</h3>
<p>Column (sample) data</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geo_id</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">count_matrix_files</span>, <span class="st">"name"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="st">"_R"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">head</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>, <span class="fl">1L</span><span class="op">)</span></span>
<span></span>
<span><span class="va">colData</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_informative</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_cols</a></span><span class="op">(</span><span class="va">count_matrix_files</span>, local.geoId <span class="op">=</span> <span class="va">geo_id</span><span class="op">)</span>,</span>
<span>        by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>samples.id <span class="op">=</span> <span class="st">"local.geoId"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">colData</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="htmlwidget-5dd5ca4d7c66cf78b984" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-5dd5ca4d7c66cf78b984">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8"],["028ec170-97d3-4e7a-a3a5-4d979e0ab553","2a688110-1dfc-4316-a9ad-f2a5d0bbc88d","3a72a97b-fd08-4ae7-82a0-4f51d5a18392","60884b69-53bc-4828-a523-bd295a04f6a5","6874fb73-f0ea-489b-8b9e-561bc7f947bb","93886610-d153-452a-b2bf-64f78a6a4065","bcec48b7-4af7-4b61-a277-343e34b59ad3","d69de99e-1b80-4ab1-8343-628b928aa124"],["2022-07-19T15:22:43.295000Z","2022-07-19T15:22:43.363000Z","2022-07-19T15:22:43.319000Z","2022-07-19T15:22:43.324000Z","2022-07-19T15:22:43.375000Z","2022-07-19T15:22:43.356000Z","2022-07-19T15:22:43.368000Z","2022-07-19T15:22:43.301000Z"],["2022-07-18T23:59:08.748000Z","2022-07-18T23:59:08.823000Z","2022-07-18T23:59:08.780000Z","2022-07-18T23:59:08.796000Z","2022-07-18T23:59:08.848000Z","2022-07-18T23:59:08.810000Z","2022-07-18T23:59:08.835000Z","2022-07-18T23:59:08.763000Z"],["2022-07-19T15:22:43.295000Z","2022-07-19T15:22:43.363000Z","2022-07-19T15:22:43.319000Z","2022-07-19T15:22:43.324000Z","2022-07-19T15:22:43.375000Z","2022-07-19T15:22:43.356000Z","2022-07-19T15:22:43.368000Z","2022-07-19T15:22:43.301000Z"],[["normal"],["erectile dysfunction"],["normal"],["erectile dysfunction"],["diabetes","erectile dysfunction"],["erectile dysfunction"],["diabetes","erectile dysfunction"],["normal"]],["donor_1","donor_6","donor_3","donor_4","donor_8","donor_5","donor_7","donor_2"],[["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["Count matrix","Featue table","UMI barcode","DNA sequence"],["DNA sequence","Count matrix","Featue table","UMI barcode"]],[[1,2],[1,2],[1,2],[1,2],[1,2],[1,2],[1,2],[8,1]],[["tar","fastq.gz"],["tar","fastq.gz"],["tar","fastq.gz"],["tar","fastq.gz"],["tar","fastq.gz"],["tar","fastq.gz"],["tar","fastq.gz"],["fastq.gz","tar"]],[[177192960,47340121901],[177192960,47509622626],[177192960,49977055324],[177192960,47042022533],[177192960,38676157359],[177192960,45056895386],[177192960,57923201058],[45298785862,177192960]],["normal","erectile dysfunction","normal","erectile dysfunction","erectile dysfunction","erectile dysfunction","erectile dysfunction","normal"],["GSM6255907","GSM6255912","GSM6255909","GSM6255910","GSM6255914","GSM6255911","GSM6255913","GSM6255908"],["normal","erectile dysfunction","normal","erectile dysfunction","erectile dysfunction","erectile dysfunction","erectile dysfunction","normal"],["GSM6255907","GSM6255912","GSM6255909","GSM6255910","GSM6255914","GSM6255911","GSM6255913","GSM6255908"],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null],[null,null,null,null,null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>entryId<\/th>\n      <th>dates.lastModifiedDate<\/th>\n      <th>dates.submissionDate<\/th>\n      <th>dates.updateDate<\/th>\n      <th>donorOrganisms.disease<\/th>\n      <th>donorOrganisms.id<\/th>\n      <th>fileTypeSummaries.contentDescription<\/th>\n      <th>fileTypeSummaries.count<\/th>\n      <th>fileTypeSummaries.format<\/th>\n      <th>fileTypeSummaries.totalSize<\/th>\n      <th>samples.disease<\/th>\n      <th>samples.id<\/th>\n      <th>specimens.disease<\/th>\n      <th>specimens.id<\/th>\n      <th>fileId<\/th>\n      <th>name<\/th>\n      <th>fileFormat<\/th>\n      <th>size<\/th>\n      <th>version<\/th>\n      <th>projectTitle<\/th>\n      <th>projectId<\/th>\n      <th>url<\/th>\n      <th>local.fileName<\/th>\n      <th>local.filePath<\/th>\n      <th>local.cellCount<\/th>\n      <th>local.geneCount<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[18,25,26]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>Experiment-wide metadata</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">s</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="va">column_is_experimentwide</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">metadata</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## Rows: 1</span></span>
<span><span class="co">## Columns: 29</span></span>
<span><span class="co">## $ cellSuspensions.organ                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ cellSuspensions.organPart             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ donorOrganisms.biologicalSex          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "male"</span></span>
<span><span class="co">## $ donorOrganisms.developmentStage       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "human adult stage"</span></span>
<span><span class="co">## $ donorOrganisms.donorCount             <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 1</span></span>
<span><span class="co">## $ donorOrganisms.genusSpecies           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Homo sapiens"</span></span>
<span><span class="co">## $ fileTypeSummaries.fileSource          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "GEO"</span></span>
<span><span class="co">## $ fileTypeSummaries.isIntermediate      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ projects.estimatedCellCount           <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> 37892</span></span>
<span><span class="co">## $ projects.laboratory                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Human Cell Atlas Data Coordinat…</span></span>
<span><span class="co">## $ projects.projectId                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "5b910a43-7fb5-4ea7-b9d6-43dbd1b…</span></span>
<span><span class="co">## $ projects.projectShortname             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "AtlasOfTheHumanCorpusCavernosum"</span></span>
<span><span class="co">## $ projects.projectTitle                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Single-cell Transcriptome Atlas…</span></span>
<span><span class="co">## $ protocols.instrumentManufacturerModel <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "Illumina NovaSeq 6000"</span></span>
<span><span class="co">## $ protocols.libraryConstructionApproach <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "10x 3' v3"</span></span>
<span><span class="co">## $ protocols.nucleicAcidSource           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "single cell"</span></span>
<span><span class="co">## $ protocols.pairedEnd                   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> FALSE</span></span>
<span><span class="co">## $ protocols.workflow                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "analysis_protocol_1"</span></span>
<span><span class="co">## $ samples.accessible                    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span> TRUE</span></span>
<span><span class="co">## $ samples.effectiveOrgan                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ samples.organ                         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ samples.organPart                     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ samples.sampleEntityType              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimens"</span></span>
<span><span class="co">## $ samples.source                        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span>
<span><span class="co">## $ sources.sourceId                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "3ab24677-3fed-42cd-b686-b82b294…</span></span>
<span><span class="co">## $ sources.sourceSpec                    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "tdr:datarepo-465c9e6a:snapshot/…</span></span>
<span><span class="co">## $ specimens.organ                       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "penis"</span></span>
<span><span class="co">## $ specimens.organPart                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "corpus cavernosum penis"</span></span>
<span><span class="co">## $ specimens.source                      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> "specimen_from_organism"</span></span></code></pre></div>
<p>Assemble into SingleCellExperiment (to come)</p>
</div>
</div>
<div class="section level2">
<h2 id="anomalies">Anomalies<a class="anchor" aria-label="anchor" href="#anomalies"></a>
</h2>
</div>
<div class="section level2">
<h2 class="unnumbered" id="sessioninfo">sessionInfo()<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.2.1 Patched (2022-10-03 r83008)</span></span>
<span><span class="co">## Platform: aarch64-apple-darwin21.6.0 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Users/ma38727/bin/R-4-2-branch/lib/libRblas.dylib</span></span>
<span><span class="co">## LAPACK: /Users/ma38727/bin/R-4-2-branch/lib/libRlapack.dylib</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] dplyr_1.0.10              hca_1.5.7                </span></span>
<span><span class="co">## [3] hcaCaseStudies_0.0.0.9005 BiocStyle_2.25.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] Rcpp_1.0.9          lattice_0.20-45     tidyr_1.2.1        </span></span>
<span><span class="co">##  [4] assertthat_0.2.1    rprojroot_2.0.3     digest_0.6.29      </span></span>
<span><span class="co">##  [7] utf8_1.2.2          mime_0.12           BiocFileCache_2.5.0</span></span>
<span><span class="co">## [10] R6_2.5.1            RSQLite_2.2.18      evaluate_0.16      </span></span>
<span><span class="co">## [13] httr_1.4.4          pillar_1.8.1        rlang_1.0.6        </span></span>
<span><span class="co">## [16] curl_4.3.2          miniUI_0.1.1.1      whisker_0.4        </span></span>
<span><span class="co">## [19] jquerylib_0.1.4     blob_1.2.3          Matrix_1.5-1       </span></span>
<span><span class="co">## [22] DT_0.25             rmarkdown_2.16      pkgdown_2.0.6      </span></span>
<span><span class="co">## [25] textshaping_0.3.6   desc_1.4.2          readr_2.1.3        </span></span>
<span><span class="co">## [28] stringr_1.4.1       htmlwidgets_1.5.4   bit_4.0.4          </span></span>
<span><span class="co">## [31] shiny_1.7.2         compiler_4.2.1      httpuv_1.6.6       </span></span>
<span><span class="co">## [34] xfun_0.33           pkgconfig_2.0.3     systemfonts_1.0.4  </span></span>
<span><span class="co">## [37] htmltools_0.5.3     tidyselect_1.1.2    tibble_3.1.8       </span></span>
<span><span class="co">## [40] bookdown_0.29       fansi_1.0.3         tzdb_0.3.0         </span></span>
<span><span class="co">## [43] dbplyr_2.2.1        later_1.3.0         rappdirs_0.3.3     </span></span>
<span><span class="co">## [46] grid_4.2.1          jsonlite_1.8.2      xtable_1.8-4       </span></span>
<span><span class="co">## [49] lifecycle_1.0.2     DBI_1.1.3           magrittr_2.0.3     </span></span>
<span><span class="co">## [52] cli_3.4.1           stringi_1.7.8       cachem_1.0.6       </span></span>
<span><span class="co">## [55] fs_1.5.2            promises_1.2.0.1    bslib_0.4.0        </span></span>
<span><span class="co">## [58] ellipsis_0.3.2      filelock_1.0.2      ragg_1.2.3         </span></span>
<span><span class="co">## [61] generics_0.1.3      vctrs_0.4.2         tools_4.2.1        </span></span>
<span><span class="co">## [64] bit64_4.0.5         glue_1.6.2          purrr_0.3.5        </span></span>
<span><span class="co">## [67] crosstalk_1.2.0     hms_1.1.2           fastmap_1.1.0      </span></span>
<span><span class="co">## [70] yaml_2.3.5          BiocManager_1.30.18 memoise_2.0.1      </span></span>
<span><span class="co">## [73] knitr_1.40          sass_0.4.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
